src/GraphQL.MicrosoftDI/ScopedFieldMiddleware.cs

```diff
@@ -0,0 +1,42 @@
+using GraphQL.Instrumentation;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace GraphQL.MicrosoftDI;
+
+/// <summary>
+/// Field middleware that creates a dependency injection scope for the duration of the field resolver's execution.
+/// </summary>
+public class ScopedFieldMiddleware : IFieldMiddleware
```

Comment:
Notice that within ScopedFieldMiddleware, there is no need to adjust the context accessor (unlike DynamicSourceStreamResolver)